RealisticPlantSim/Assets/Scripts/Procedural Terrain/PlantGenerator.cs

```diff
@@ -77,17 +95,17 @@ public IEnumerator spawnPlantsInXZRange()
 
             PlantSpawnSettings spawnSettings = plantSpawnSettings[randomPlantIndex];
 
-            AbstractRandomPlacement randomPlacement;
+            AbstractPlacementStrategy placementStrategy;
 
-            if (spawnSettings.randomPlacement.GetType().IsAssignableFrom(typeof(AbstractRandomPlacement)))
+            if (spawnSettings.placementStrategy.GetClass().IsSubclassOf(typeof(AbstractPlacementStrategy)))
```

Comment:
"if(placementStrategy is AbstractPlacementStrategy)" should also work here. also if placementStrategy is passed through the inspector you don't need to check whether its of the Abstract type, unity will only accept AbstractPlacementStrategy types

Reply:
Your first point was valid. But the second one is not true. Unfortunately i need to use the type MonoScript to be able to drag a script into the inspector. If i use the abstract class you cant drag in a script.

Reply:
you can make unity only accept `AbstractPlacementStrategy` scripts, but it requires editor code

Reply:
Ill take a look into that